blog/chakra-ui-the-hidden-treasures.md

```diff
@@ -0,0 +1,79 @@
+
+### The hooks
+
+The first package I try to looked into was the [@chakra-ui/hooks](https://github.com/chakra-ui/chakra-ui/tree/main/packages/hooks) and I found the first treasures:
+
+The [useOutsideClick](https://chakra-ui.com/docs/hooks/use-outside-click), a common one to hide popovers when the user clicks outside them. And then the [useMergeRefs](https://chakra-ui.com/docs/hooks/use-merge-refs) when you need to merge several react refs into one. I contributed with doc pages for both hooks. 😁 You can find more [useful](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/src/use-const.ts) [hooks](https://github.com/chakra-ui/chakra-ui/blob/main/packages/hooks/src/use-why-update.ts).
+
+Like this special one: the [usePopper](https://github.com/chakra-ui/chakra-ui/blob/main/packages/popper/README.md), a wrapper for the [popper.js](https://popper.js.org/) 🍿 library, converting it into a magic hook. This is an internal hook and it's not documented in the website, but is still exported and you can use it, but ⚠️ it's not covered by [semver](https://semver.org/). 🙀
+
+### The utils
+
+Another internal package is the [@chakra-ui/utils](https://github.com/chakra-ui/chakra-ui/tree/main/packages/utils).
+
+You can find functions for [arrays](https://github.com/chakra-ui/chakra-ui/blob/main/packages/utils/src/array.ts) that remembers to the Lodash library. Also to manage [functions](https://github.com/chakra-ui/chakra-ui/blob/main/packages/utils/src/function.ts) and parse [numbers](https://github.com/chakra-ui/chakra-ui/blob/main/packages/utils/src/number.ts).
+
+Another one to work with [objects](https://github.com/chakra-ui/chakra-ui/blob/main/packages/utils/src/object.ts) with [`lodash.mergewith`](https://www.npmjs.com/package/lodash.mergewith) as a dependency.
+
+You can use [`addDomEvent`](https://github.com/chakra-ui/chakra-ui/blob/main/packages/utils/src/dom.ts#L41) to add global event listeners without leaks.
+
+```jsx
+import {addDomEvent} from '@chakra-ui/utils';
+
+const MyNiceFeature = () => {
+  const [height, setHeight] = useState(window.innerHeight - 50);
+  useEffect(
+    () =>
+      addDomEvent(window, 'resize', () =>
+        setHeight(window.innerHeight - 50),
+      ),
+    [],
+  );
+  ...
+};
+```
+
+If this helpers are already loaded in your bundle, what if you don't repeat the code and you reuse this in your React app code.
+
+### The React code
+
+I found myself reading the code by [Sage](https://github.com/readme/segun-adebayo) and the contributors and learning another way to handle heavy UI components, like the [@chakra-ui/menu](https://github.com/chakra-ui/chakra-ui/tree/main/packages/menu). Using hooks and event handlers to handle the user interactions in a fluent way. 🤯
```

Comment:
Maybe it has to do with that I don't know what Sage is o exactly what are you talking about, but this paragraph seems a bit confusing to me.